lib/upload.js

```diff
@@ -262,6 +243,33 @@ class Upload {
     }
   }
 
+  _isOnline() {
+    let isOnline = true;
+    if (typeof window !== "undefined" &&
+        "navigator" in window &&
+        window.navigator.onLine === false) {
+      isOnline = false;
+    }
+
+    return isOnline;
+  }
+
+  _shouldRetry(err) {
+    // We only attempt a retry if
+    // - retryDelays option is set
+    // - we didn't exceed the maxium number of retries, yet, and
+    // - this error was caused by a request or it's response and
+    // - the error is server error (i.e. no a status 4xx or a 409 or 423) and
+    // - the browser does not indicate that we are offline
+    let status = err.originalRequest ? err.originalRequest.status : 0;
+    let isServerError = !inStatusCategory(status, 400) || status === 409 || status === 423;
+    return this.options.retryDelays != null &&
+                      this._retryAttempt < this.options.retryDelays.length &&
+                      err.originalRequest != null &&
+                      isServerError &&
+                      this._isOnline();
+  }
+
   abort(shouldTerminate, cb) {
```

Comment:
Please also update the README to explain that the termination request is retried.